sql-plugin/src/main/spark350/scala/org/apache/spark/rapids/shims/ShuffleExchangeLikeShim.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "350"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.rapids.shims
+
+import org.apache.spark.sql.execution.exchange.ShuffleExchangeLike
+
+trait ShuffleExchangeLikeShim extends ShuffleExchangeLike {
+  override def advisoryPartitionSize: Option[Long] = None
```

Comment:
Don't we want to override this when running on Spark 3.5?  Rather than shimming this class/method, I'd rather see GpuShuffleExchange act like the 3.5 code where it adds a new constructor argument, advisoryPartitionSize that defaults to None.  Spark 3.5 would shim the meta for ShuffleExchangeExec to specify this additional constructor argument (by copying the value from the CPU node it is replacing) while Spark 3.4 and earlier would not provide the argument, leaving it to None.